Feat: Landing page

KarinaDavtyan opened the pull request:
Summary:
+ Added initial layout for landing page

+ Responsiveness will be improved

+ The design will be changed

Closes #174

tbergq commented on .env-sample:
```diff
@@ -1 +0,0 @@
-GRAPHQL_URL=
```
why is this file deleted?

KarinaDavtyan replied:
Sorry misread initial configuration instructions, renamed files without doing a copy first

I added them back in a new commit

tbergq commented on apps/graphql/.env-example:
```diff
@@ -1,3 +0,0 @@
-API_KEY=
```
why is it deleted?

KarinaDavtyan replied:
Same⬆️

tbergq commented on apps/landingPage/package.json:
```diff
@@ -0,0 +1,53 @@
+{
+  "name": "gatsby-starter-default",
```
maybe we should name this package differently?

KarinaDavtyan replied:
Just changed with commit

tbergq commented on apps/landingPage/package.json:
```diff
@@ -0,0 +1,53 @@
+    "react": "^16.6.3",
+    "react-dom": "^16.6.3",
+    "react-helmet": "^5.2.0",
+    "styled-components": "^4.1.3"
```
could you move dependencies that are already included in other `package.json` to peerDependency? like `@kiwicom/orbit-components` and `react`

KarinaDavtyan replied:
Moved in the latest commit

tbergq commented on apps/landingPage/package-lock.json:
```diff
@@ -0,0 +1,15743 @@
+{
```
could we stick to having `yarn.lock` only please. No package-lock.json 😊

KarinaDavtyan replied:
Ok! never again

tbergq commented on apps/landingPage/package.json:
```diff
@@ -0,0 +1,46 @@
+{
+  "name": "Margarita's landing page",
```
Sorry, I should have mentioned in latest commit,  but can we keep naming convention. `@kiwicom/margarita-landing-page`

KarinaDavtyan replied:
Will fix now, and should i keep description blanked as it is now?

tbergq commented on apps/landingPage/src/components/demo.js:
```diff
@@ -0,0 +1,64 @@
+  </div>
+)
+
+const styles = {
```
Also I see `styled-components` as one of the dependencies. Could we use that for styling instead of inline-styles?

KarinaDavtyan replied:
Ok! Changed all inline styles into styled components

tbergq commented on apps/landingPage/src/styles/global.css:
```diff
@@ -0,0 +1,4 @@
+body {
+  margin: 0;
+  -webkit-font-smoothing: antialiased;
+}
```
Also this should be replace with [styled-components](https://www.styled-components.com/docs/api#createglobalstyle)

KarinaDavtyan replied:
Sorry missed that, fixed now

tbergq commented on apps/landingPage/src/components/demo.js:
```diff
@@ -0,0 +1,99 @@
+
+type Props = {
+  iframeWidth: number,
+  iframeHeight: number,
```
Could this be exact type with read-only props?
```
type Props = {|
  +iframeWidth: number,
  +iframeHeight: number,
|};
```

tbergq commented on apps/landingPage/src/components/headline.js:
```diff
@@ -0,0 +1,82 @@
+
+type Props = {
+  iframeWidth: number,
+};
```
also please exact and read-only 😊

tbergq commented on apps/landingPage/src/components/info.js:
```diff
@@ -0,0 +1,136 @@
+
+type Props = {
+  iframeWidth: number,
+};
```
read-only and exact

RobinCsl commented:
Once it's ready, let me know and we can have it on auto-deploy on Netlify too.

KarinaDavtyan commented on apps/landingPage/src/components/footer.js:
```diff
@@ -0,0 +1,90 @@
+  margin-top: 3vh;
+  height: 10vh;
+  align-items: center;
+  border-top: 2px solid #f5f7f9;
```
Changed to separator

tbergq commented:
LGTM, but I think also @RobinCsl could review it 😊 

Also please squash your commits and rebase with master.

FilipMessa commented on apps/landingPage/gatsby-config.js:
```diff
@@ -0,0 +1,36 @@
+    },
+    // this (optional) plugin enables Progressive Web App + Offline functionality
+    // To learn more, visit: https://gatsby.app/offline
+    // 'gatsby-plugin-offline',
```
Do we need this comment?

FilipMessa commented on apps/landingPage/package.json:
```diff
@@ -0,0 +1,46 @@
+  "name": "@kiwicom/margarita-landing-page",
+  "private": true,
+  "version": "0.1.0",
+  "dependencies": {
```
I realized that there are few gatsby plugins which are not used anywhere, can you check and clean them.

FilipMessa commented on apps/landingPage/package.json:
```diff
@@ -0,0 +1,46 @@
+    "gatsby-plugin-sharp": "^2.0.19",
+    "gatsby-source-filesystem": "^2.0.8",
+    "gatsby-transformer-sharp": "^2.1.13",
+    "prop-types": "^15.6.2",
```
I think this package also does not use anywhere.

KarinaDavtyan replied:
There were too many conflicts in yarn.lock

KarinaDavtyan replied:
which one? prop-types?

FilipMessa commented:
What was the reason to remove `yarn.lock`?

FilipMessa commented on apps/landingPage/package.json:
```diff
@@ -0,0 +1,41 @@
+    "@kiwicom/orbit-components": "^0.28.2",
+    "@kiwicom/orbit-design-tokens": "^0.3.0",
+    "gatsby": "^2.0.105",
+    "gatsby-image": "^2.0.20",
```
The `gatsby-image` is nowhere to be used. Or maybe I did not find where?

KarinaDavtyan replied:
You are right, its also not used

FilipMessa commented:
BTW maybe will be nice use `gatsby-image` in future because of the images optimization, but yes it isn't part of this PR.

https://medium.com/@kyle.robert.gill/ridiculously-easy-image-optimization-with-gatsby-js-59d48e15db6e

michalsanger commented:
How can I run it? The readme looks like default Gatsby (plz fix). 

I tried `yarn gatsby develop` but that does not work in root or `apps/landingPage`

KarinaDavtyan commented:
> How can I run it? The readme looks like default Gatsby (plz fix).
> 
> I tried `yarn gatsby develop` but that does not work in root or `apps/landingPage`

Fixed now.
Added `yarn landing-page` from the root folder
and `yarn start` from apps/landingPage
Also fixed readme

RobinCsl commented on apps/landingPage/src/components/footer.js:
```diff
@@ -0,0 +1,97 @@
+// $FlowFixMe - Cannot resolve module RelativeImageStub ( possible solution https://bit.ly/2BYfHRr )
+import logo from '../images/logo.png';
+
+const links = [
```
These could be extract to a config file in the future, because I believe we will re-use these links throughout the page.
It's just a minor improvement, and could be done later.

RobinCsl commented on apps/landingPage/src/components/header.js:
```diff
@@ -0,0 +1,67 @@
+    </a>
+    <HeaderRight>
+      {links.map(el => {
+        const internal = /^\/(?!\/)/.test(el.link);
```
Is that necessary? Does Gatsby Link only work for internal links?

KarinaDavtyan replied:
Yes it works only with internal, but anyway I think I ll change this part

RobinCsl commented on apps/landingPage/src/components/headline.js:
```diff
@@ -0,0 +1,82 @@
+  display: flex;
+  height: 40vh;
+  margin-bottom: 4vh;
+  background-image: url(https://tequila.kiwi.com/portal/static/media/airport-runway.4acb8968.jpg);
```
Are we sure this hash in the filename will never change? That might break the site. One solution could be to extract it and add it to the repo. Although I am not sure this the best approach either.
@tbergq thoughts?

KarinaDavtyan replied:
Sorry I couldnt find the image and temporally take it from tequila page

tbergq replied:
You know you can right click on the image in your browser, then click save image as...

KarinaDavtyan replied:
Didnt think of that😅
I ll add it to the repo in next pr then

RobinCsl commented on apps/landingPage/src/components/info.js:
```diff
@@ -0,0 +1,136 @@
+      <ContainerInfoBottom width={iframeWidth}>
+        <Info>
+          <Text>
+            {' '}
```
Is that space necessary?

KarinaDavtyan replied:
It was auto fix, I guess. Will remove it when changing to new design

RobinCsl commented on apps/landingPage/src/components/info.js:
```diff
@@ -0,0 +1,136 @@
+        </Info>
+        <Info>
+          <Text>
+            {' '}
```
Idem

RobinCsl commented on apps/landingPage/src/components/info.js:
```diff
@@ -0,0 +1,136 @@
+        </Info>
+        <Info>
+          <Text>
+            {' '}
```
Idem

RobinCsl commented on apps/landingPage/src/components/playground.js:
```diff
@@ -0,0 +1,38 @@
+      title="GraphQL playground"
+      width={iframeWidth}
+      height={iframeHeight}
+      src="https://kiwi-graphiql.now.sh/"
```
```suggestion
      src="https://p2kwd3i3a8.execute-api.eu-central-1.amazonaws.com/staging/graphql"
```
